### DISCUSSION TITLE
Internal Server Error with RENAL substructure segmentation

### DISCUSSION THREAD

**Discussion by RotoDrache** [Project-MONAI/MONAILabel#1235] [Q&A]
I am trying to use the **Use Case 2: Bundle with Customized Scripts for Renal Substructure Segmentation in the QUICKSTART menu to segment the kidney.**

I have installed monailabel in conda env. and have followed the steps in the quickstart but am getting the following error when I try to run auto segmentaton

Traceback (most recent call last):
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/NA-MIC/Extensions-31480/MONAILabel/lib/Slicer-5.3/qt-scripted-modules/MONAILabel.py", line 1496, in onClickSegmentation
result_file, params = self.logic.infer(model, image_file, params, session_id=self.getSessionId())
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/NA-MIC/Extensions-31480/MONAILabel/lib/Slicer-5.3/qt-scripted-modules/MONAILabel.py", line 2231, in infer
result_file, params = client.infer(model, image_in, params, label_in, file, session_id)
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/NA-MIC/Extensions-31480/MONAILabel/lib/Slicer-5.3/qt-scripted-modules/MONAILabelLib/client.py", line 199, in infer
raise MONAILabelException(
MONAILabelLib.client.MONAILabelException: (2, "Status: 500; Response: b'Internal Server Error'")

Any help would be most appreciated

**Comment by tangy5** [comment]
Hi @RotoDrache , I didn't see this issue before, looking at the logs, it seems the plugin has a connection issue with MONAI Label server. 
**Can you try to re-install Slicer Plugin?** , Slicer Plugin had changes in the most recent release.
I see you are using latest Slicer 5.3, make sure you are using latest MONAI Label (version 0.6.0) as well. 
For debugging purpose, you can also check whether the host and port are matched between Slicer and monailabel server.
Hope this helps.

**Comment by RotoDrache** [comment]
**Your suggestions seemed to have corrected the posted issue so many thanks but now I'm running into new error related to gpu memory I think**
Traceback (most recent call last):
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/MONAILabel/plugins/slicer/MONAILabel/MONAILabel.py", line 1496, in onClickSegmentation
result_file, params = self.logic.infer(model, image_file, params, session_id=self.getSessionId())
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/MONAILabel/plugins/slicer/MONAILabel/MONAILabel.py", line 2231, in infer
result_file, params = client.infer(model, image_in, params, label_in, file, session_id)
File "/home/sam/Desktop/Slicer-5.3.0-2022-12-26-linux-amd64/MONAILabel/plugins/slicer/MONAILabel/MONAILabelLib/client.py", line 199, in infer
raise MONAILabelException(
MONAILabelLib.client.MONAILabelException: (2, "Status: 500; Response: b'Internal Server Error'")

**the error that appears is** 
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 864.00 MiB (GPU 0; 3.92 GiB total capacity; 1.89 GiB already allocated; 626.00 MiB free; 2.35 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

**My setup is modest** 
1. Intel® Xeon(R) CPU E5-1607 v4 @ 3.10GHz × 4
2. NVIDIA Corporation GP107 [GeForce GTX 1050 Ti]
3. Cuda 11.8

**Any suggestions most appreciated**

**Reply by diazandr3s** [reply]
Hi @RotoDrache,

From the logs, I can see the GPU you're using isn't big enough to run inference on this image size.
Please try with a bigger GPU - an 8GB GPU should be good.

Hope this helps,

**Reply by RotoDrache** [reply]
Thank-you for your reply. I did come to that conclusion. I rebuilt a workstation around a 12 GB GPU and all is good. Also I am very keen to work on a model for renal pathology, tumors (bengn and malignant) and cysts. Can you direct me to someone that I may be able to collaborate with?

**Reply by diazandr3s** [reply]
Sure! Please register for this workshop. We're presenting the latest of MONAI Label: https://projectweek.na-mic.org/PW38_2023_GranCanaria/MONAILabel_Workshop.html
It is free, you could also learn from others and present the work you're doing.
The Slicer week is happening at the end of Jan. I'd recommend you participate. It is also free and you can participate online: https://projectweek.na-mic.org/PW38_2023_GranCanaria/
